packages/core/useResize/index.ts

```diff
@@ -0,0 +1,336 @@
+    )
+      setCursorAndDirection('nwse-resize', 'top-left')
+
+    else if (
+      ((currentY - top) < 2 || (currentX - right) > -2)
+      && Math.abs(currentY - top) < 8
+      && Math.abs(currentX - right) < 8
+      && isEdgeActive('top-right')
+      && isOnForeground(right, top, true, -1, 1)
+    )
+      setCursorAndDirection('nesw-resize', 'top-right')
+
+    else if (
+      ((currentY - bottom) > -2 || (currentX - left) < 2)
+      && Math.abs(currentY - bottom) < 8
+      && Math.abs(currentX - left) < 8
+      && isEdgeActive('bottom-left')
+      && isOnForeground(left, bottom, true, 1, -1)
+    )
+      setCursorAndDirection('nesw-resize', 'bottom-left')
+
+    else if (
+      ((currentY - bottom) > -2 || (currentX - right) > -2)
+      && Math.abs(currentY - bottom) < 8
+      && Math.abs(currentX - right) < 8
+      && isEdgeActive('bottom-right')
+      && isOnForeground(right, bottom, true, -1, -1)
+    )
+      setCursorAndDirection('nwse-resize', 'bottom-right')
+
+    else if (
+      (currentY - bottom) < 8
+      && (currentY - bottom) >= 0
+      && currentX > left
+      && currentX < right
+      && isEdgeActive('bottom')
+      && isOnForeground(currentX, bottom)
+    )
+      setCursorAndDirection('ns-resize', 'bottom')
+
+    else if (
+      (currentY - top) > -8
+      && (currentY - top) <= 0
+      && currentX > left
+      && currentX < right
+      && isEdgeActive('top')
+      && isOnForeground(currentX, top)
+    )
+      setCursorAndDirection('ns-resize', 'top')
+
+    else if (
+      (currentX - left) > -8
+      && (currentX - left) <= 0
+      && currentY > top
+      && currentY < bottom
+      && isEdgeActive('left')
+      && isOnForeground(left, currentY)
+    )
+      setCursorAndDirection('ew-resize', 'left')
+
+    else if (
+      (currentX - right) < 8
+      && (currentX - right) >= 0
+      && currentY > top
+      && currentY < bottom
+      && isEdgeActive('right')
+      && isOnForeground(right, currentY)
+    )
+      setCursorAndDirection('ew-resize', 'right')
+
+    else
+      setCursorAndDirection('', '', '')
+  }
+
+  function setSize(width: number, height: number) {
+    target.value!.style.width = `${clamp(width, unref(minWidth) as number, unref(maxWidth) as number)}px`
+    target.value!.style.height = `${clamp(height, unref(minHeight) as number, unref(maxHeight) as number)}px`
+  }
+
+  function useElementSize() {
+    return useResizeObserver(target, () => {
+      widthRef.value = target.value!.getBoundingClientRect().width
+      heightRef.value = target.value!.getBoundingClientRect().height
+    }).stop
+  }
+
+  tryOnScopeDispose(stop)
+
+  return {
+    width: widthRef,
+    height: heightRef,
+    stop,
+    start,
+    direction,
+    isActive,
+    isOverEdge,
+    isResizing,
+    onResizeStart: onResizeStart.on,
+    onResizeMove: onResizeMove.on,
+    onResizeEnd: onResizeEnd.on,
+  }
+}
+
+function clamp(value: number, min: number, max: number) {
```

Comment:
This can be replaced with the existing clamp function here https://github.com/vueuse/vueuse/blob/499a6987128e3db6862f57bcc189095d6c501a67/packages/shared/utils/is.ts#L18